### ISSUE TITLE
Recent PAM50 changes have caused test failures

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#4254]
While working on the lumbar tutorial, I needed to update the PAM50 link to include the new disc label (`60`).

However, doing so also added [a number of other changes](https://github.com/spinalcordtoolbox/PAM50/compare/r20230519...r20231011) introduced recently by @jcohenadad. Somewhere within these changes plus my change, there are now 2 new test failures:

1. `test_sct_label_vertebrae_high_value_warning`
    
    <details>
    
    ```
    test_sct_label_vertebrae_high_value_warning - AssertionError: assert 'Disc value not included in template.' in 'WARNING  
    ```
    https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/78d46e0d1a75dfa6a27689931bf4111a233c04f1/testing/cli/test_cli_sct_label_vertebrae.py#L70-L75
    
    What _should_ happen is that `-initz 40,19` (z=40, label=19) should trigger this warning:
    
    https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/78d46e0d1a75dfa6a27689931bf4111a233c04f1/spinalcordtoolbox/vertebrae/core.py#L208-L212
    
    (This happens because we start with `19`, then increment in the inferior direction (`20`), and then `list_disc_value_template.index(current_value)` fails with `current_value==20`)
    
    However, for some reason, this warning isn't being emitted when it should be.
    
    </details>

2. `batch_processing.sh` results changing

    ```
    =========================== short test summary info ============================
    FAILED testing/batch_processing/test_batch_processing.py::test_batch_processing_results[mt/mtr_in_wm.csv-0-MAP()] - assert 54.39463563966487 == 54.388569575185116 ± 5.4e-05
      comparison failed
      Obtained: 54.39463563966487
      Expected: 54.388569575185116 ± 5.4e-05
    FAILED testing/batch_processing/test_batch_processing.py::test_batch_processing_results[dmri/fa_in_cst.csv-0-WA()] - assert 0.7793857114020448 == 0.7793405440192693 ± 7.8e-07
      comparison failed
      Obtained: 0.7793857114020448
      Expected: 0.7793405440192693 ± 7.8e-07
    FAILED testing/batch_processing/test_batch_processing.py::test_batch_processing_results[dmri/fa_in_cst.csv-1-WA()] - assert 0.775688329824498 == 0.7755611783165834 ± 7.8e-07
      comparison failed
      Obtained: 0.775688329824498
      Expected: 0.7755611783165834 ± 7.8e-07
    ```

I will try to get to the bottom of which changes caused which failures, the fix things as appropriate.

**Comment by jcohenadad** [comment]
Thank you for looking into this @joshuacwnewton.

I'm not surprised about "`batch_processing.sh` results changing", (it is to be expected), and the changes are very small, so I wouldn't worry too much. But still better to understand the root cause

**Comment by joshuacwnewton** [comment]
## Failure 1: `test_sct_label_vertebrae_high_value_warning`

The explanation for this first test failure involves quite a lot of details (apologies in advance):

<details>

### Background

The relevant part of `sct_label_vertebrae` works as follows:

1. Load the `PAM50_levels.nii.gz` file, then infer the template disc locations using the interfaces between pairs of levels.
2. Do passes in the superior and inferior directions, repeating the steps below:
  i. Search for a disc in the subject data using correlation with the PAM50 template (using the disc locations determined above).
  ii. Compute a "correction factor" by comparing the relative disc-disc distances between the subject and the PAM50 template.
  iii. Use this correction factor to continuously update a list of estimates for where the next discs will be.
  iv. Use the estimate to determine where in the subject to search for the next disc.

### Cause of test failure

The changes to the `PAM50_levels.nii.gz` file include extending the top and bottom levels (https://github.com/spinalcordtoolbox/PAM50/pull/21). This improves the inferred locations of the top and bottom discs (`1` and `20` respectively), thus improving the estimations for where the top and bottom discs will be in the subject image. 

((Before, the estimate was much too small (16 voxels between discs `19`/`20`), due to the fact that the old PAM50 file cut off [way too early](https://user-images.githubusercontent.com/2482071/258561785-0ad2da5b-f7f1-417f-839b-f4f1a7a77c9e.gif) in the lumbar region. But now, we correctly estimate that disc `20` will be 65 voxels away from disc `19`.))

However, even though the estimate is now much more accurate, remember that we're working with `sct_testing_data`, which is cropped! Because the estimate for disc `20` increased so significantly, by the time we try to estimate the _next_ disc (`21`), we're outside the search range of the test image, and the function exits early, before the expected warning can be triggered.

</details>

tl;dr:  this "failure" is only due to the small size of the test image -- it turns out that the actual functionality is working even better now. :tada:

I don't think there's really a way that we can fix the old test without increasing the size of the test image. So, I think for now we can just skip the test, and point to this comment as the reason why. (It isn't a vital test or anything -- it only checks that a warning was raised for a very specific corner case, so I'm not too worried about skipping it.)

## Failure 2: `batch_processing.sh` results

The only results that have changed are the MT and dMRI values, meaning that the other results (e.g. T2/T2* CSA) are unchanged. This means that vertebral labeling + registration to template + `sct_process_segmentation` are unaffected. :tada:

My hunch is that these results changing are due simply to the fact that both MT/dMRI involve multimodal registration to the PAM50 template, using the `PAM50_cord.nii.gz` file directly:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/92a17701836ff09c66c365fddb7079e67ba11846/batch_processing.sh#L173-L176

Given that the cord was recently symmetrized, I'm not surprised that the registration results would change, as you suggested @jcohenadad. So, I can just go ahead and update the cached results in PR #4250.
